src/link/__tests__/index.test.tsx

```diff
@@ -158,4 +180,54 @@ describe('Link component', () => {
       );
     });
   });
+
+  describe('Analytics', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockFunnelMetrics();
+    });
+
+    test('does not send any metrics when not in a funnel context', () => {
+      const wrapper = renderLink({ href: '#', external: true });
+      wrapper.click();
+
+      expect(FunnelMetrics.externalLinkInteracted).not.toHaveBeenCalled();
+    });
+
+    test('sends a externalLinkInteracted metric when an external link is clicked within a Funnel Context', () => {
+      const { container } = render(
+        <AnalyticsFunnel funnelType="single-page" optionalStepNumbers={[]} totalFunnelSteps={1}>
+          <Link href="#" external={true} />
+        </AnalyticsFunnel>
+      );
+      const wrapper = createWrapper(container).findLink()!;
+      wrapper.click();
+
+      expect(FunnelMetrics.externalLinkInteracted).toHaveBeenCalled();
+      expect(FunnelMetrics.externalLinkInteracted).toHaveBeenCalledWith(
+        expect.objectContaining({
+          funnelInteractionId: expect.any(String),
+          elementSelector: expect.any(String),
```

Comment:
I would suggest checking for the correct strings, too.

Reply:
Changed funnelInteractionId but left elementSelector as its a generated id